interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is repeating a defined process infinitely or until a condition is met. 
 
-  Researched answer:
+  Researched answer: Iteration is the process of performing a specific process until a specific number of times or until a condition evaluates to true. If the condition always evaluates to true without a defined number of times or limit to the iteration, the process will run infinitely. 
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The .map higher order function takes in the parameters of value, index, and length. The .map function is intended to work on arrays. Only one parameter is required, but it doesn't matter which one of the three. 
 
-  Researched answer:
+  Researched answer: The .map method takes in parameters of value, index, and length of an array. The value parameter is required and the index and length parameters are optional because the .map method is a higher order function iterates through an array to give an output of a new array and an array cannot exist without defined values. The index and length of the array will be deifined by the input values and number of elements input.  
 
 
 
 3. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructering is a method of taking individual key: value pairs out of an object as independent variables. 
 
-  Researched answer:
+  Researched answer: Object destructuring is a way to assign variables by taking the properties of an object and breaking them down into individual variables that can be recalled throughout the code as a faster way to manipulate objects and nested properties.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function should be universal and can be take various inputs such as defined variables outside of the function. Methods are built into the language of javascript to be used to perform a specific task on a specific or limited data type.
 
-  Researched answer:
+  Researched answer: Functions are blueprint for logic, that are reusable pieces of code that only execute when being called on by its name and that uses an input parameter to execute. Methods are associated with object data types, is invoked using the dot notation or square bracket notation, and operates on the data contained in a class. 
```

Comment:
Methods are functions that belong to a obj (generally one assoc with a class) and act upon the object they belong too. (i.e. string methods act upon anything made with the class of string)